app/src/main/java/com/example/tsundokun/ui/home/HomeScreen.kt

```diff
@@ -1,17 +1,418 @@
+    if (visible.value) {
+        Card(
+            modifier = modifier
+                .padding(16.dp)
+                .border(width = 1.dp, color = Color.Gray, shape = RoundedCornerShape(8.dp)),
+            colors = CardDefaults.cardColors(
+                containerColor = MaterialTheme.colorScheme.background,
+            ),
+        ) {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Image(
+                    painter = painterResource(R.drawable.tsundokun),
+                    contentDescription = "",
+                    modifier = Modifier
+                        .padding(horizontal = 8.dp)
+                        .weight(1f),
+                )
+                Column(
+                    modifier = Modifier
+                        .padding(8.dp)
+                        .weight(3f),
+                ) {
+                    Row(verticalAlignment = Alignment.CenterVertically) {
+                        Text(
+                            text = stringResource(id = R.string.tsundokun_report, 10),
+                            style = MaterialTheme.typography.bodyLarge,
+                            modifier = modifier.weight(10f),
+                        )
+                        IconButton(onClick = { visible.value = false }) {
+                            Icon(
+                                imageVector = Filled.Close,
+                                contentDescription = stringResource(R.string.button_close),
+                                Modifier
+                                    .width(16.dp)
+                                    .weight(1f),
+                            )
+                        }
+                    }
+                    Text(
+                        text = stringResource(id = R.string.tsundokun_report_detail),
+                        style = MaterialTheme.typography.bodySmall,
+                        modifier = modifier.padding(bottom = 8.dp),
+                    )
+                }
+            }
+        }
+    }
+}
+
+/*
+ * つんどくんのレポートを表示するカードのプレビュー
+ */
+@Preview
+@Composable
+private fun TsundokunReportPreview() {
+    TsundokunReport()
+}
+
+/*
+ * Webページのリスト
+ * タブで表示を切り替えられる
+ */
+@Composable
+fun WebPageListScreen() {
+    var tabName = mutableMapOf("ALL" to "すべて", "FAVORITE" to "お気に入り")
+    var tabSelected by rememberSaveable { mutableStateOf(Screen.ALL) }
+    Column {
+        TabRow(
+            selectedTabIndex = tabSelected.ordinal,
+        ) {
+            Screen.values().map { it.name }.forEachIndexed { index, title ->
+                Tab(
+                    text = { Text(text = tabName[title].toString()) },
+                    selected = tabSelected.ordinal == index,
+                    onClick = { tabSelected = Screen.values()[index] },
+                )
+            }
+        }
+        when (tabSelected) {
+            Screen.ALL -> AllScreen()
+            Screen.FAVORITE -> FavoriteScreen()
```

Comment:
`AllScreen()` と `FavoriteScreen()` というcomposableを作ってありますが、`WebPageList()` を直接呼び出す(& 実際には渡す引数を変える) 方が、スケールできる気がしますが、いかがでしょうか？
```suggestion
            Screen.ALL -> WebPageList(webPageList = allTsundokus)
            Screen.FAVORITE -> WebPageList(webPageList = favoriteTsundoku)
```